src/v/raft/types.cc

```diff
@@ -249,6 +251,162 @@ std::ostream& operator<<(std::ostream& o, const install_snapshot_reply& r) {
     return o;
 }
 
+ss::future<> heartbeat_request::serde_async_write(iobuf& dst) {
+    vassert(!heartbeats.empty(), "cannot serialize empty heartbeats request");
+
+    struct sorter_fn {
+        constexpr bool operator()(
+          const raft::heartbeat_metadata& lhs,
+          const raft::heartbeat_metadata& rhs) const {
+            return lhs.meta.commit_index < rhs.meta.commit_index;
+        }
+    };
+
+    iobuf out;
+    auto& request = *this;
+
+    std::sort(
+      request.heartbeats.begin(), request.heartbeats.end(), sorter_fn{});
+
+    co_await ss::coroutine::maybe_yield();
```

Comment:
question, is this trying to yield back to the reactor because `std::sort` may take a while? So if we had something like `ss::future<> async_sort()` you'd want to use that instead?

Reply:
Yeah that was just for arguments sake but I see your point here. Why not use an async for loop in the proceeding loop below this? Do we expect the number of iterations to be small?

Reply:
is the yield-in-loop you are thinking about in the latest force push?

Reply:
Yes, so the question is, what advantage does
```
for(...) {
  co_await ss::coroutine::maybe_yield;
}
```
Have over
```
ss::do_for_each(...); 
```
My immediate guess is its just easier to write `maybe_yield` within the context of coroutines